app/models/charge.js

```diff
@@ -89,26 +89,20 @@ module.exports = correlationId => {
     return response.body
   }
 
-  const patch = function (chargeId, op, path, value, subSegment) {
-    return new Promise(function (resolve, reject) {
+  const patch = async function (chargeId, op, path, value, subSegment) {
+    try {
       const payload = {
         op: op,
         path: path,
         value: value
       }
-      connectorClient({ correlationId }).patch({ chargeId, payload }, subSegment)
-        .then(response => {
-          const code = response.statusCode
-          if (code === 200) {
-            resolve()
-          } else {
-            reject(new Error('Calling connector to patch a charge returned an unexpected status code'))
-          }
-        })
-        .catch(err => {
-          reject(err)
-        })
-    })
+      const response = await connectorClient({ correlationId }).patch({ chargeId, payload }, subSegment)
+      if (response.statusCode !== 200) {
+        throw new Error('Calling connector to patch a charge returned an unexpected status code')
+      }
+    } catch (err) {
+      throw new Error('CLIENT_UNAVAILABLE', err)
```

Comment:
Is this behaviour not different now by passing `CLIENT_UNAVAILABLE`? Without good unit tests this might cause an issue with upstream handling, no idea though.

Reply:
the old code did
```
        .catch(err => {	
          reject(err)	
        })
```
would it better to just do `throw err` instead of `throw new Error('CLIENT_UNAVAILABLE', err)` here?
It's going to get caught in charge_controller.js like so:
```
      try {
        await Charge(req.headers[CORRELATION_HEADER]).patch(req.chargeId, 'replace', 'email', userEmail)
      } catch (err) {
        return responseRouter.response(req, res, 'SYSTEM_ERROR', withAnalytics(charge))
      }
```

Reply:
I think keeping the same behaviour would be good for something that's solely a refactor. 

My understanding is that if you remove the try/ catch entirely it will behave the same (as `err => { reject(err) }`)

Reply:
@sfount ah, I didn't know that. I've removed the try catch now.